src/main/java/org/osc/controller/nsc/restserver/api/InspectionApis.java

```diff
@@ -0,0 +1,403 @@
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport/{insportId}")
+    @GET
+    public InspectionPortEntity getInspectionPortElement(@PathParam("ipAddr") String ipAddr,
+            @PathParam("insportId") String insportId) throws Exception {
+        if (insportId == null) {
+            throw new IllegalArgumentException("Attempt to retrive null inspection port");
+        }
+        LOG.info(String.format("Getting the inspection port element for id %s ", "" + insportId));
+        init(ipAddr);
+        InspectionPortEntity inspectionPort = new InspectionPortEntity(insportId, null, null);
+        return (InspectionPortEntity) this.api.getInspectionPort(inspectionPort);
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport/{insportId}/inshook")
+    @POST
+    public String createInspectionHook(@PathParam("ipAddr") String ipAddr, @PathParam("insportId") String insportId,
+            InspectionHookEntity entity)
+                    throws Exception {
+
+        if (insportId == null) {
+            throw new IllegalArgumentException("Attempt to create inspection hook with null inspection port");
+        }
+        LOG.info(String.format("Creating inspection hook with inspection port id %s", "" + insportId));
+        init(ipAddr);
+        InspectionPortEntity inspectionPort = new InspectionPortEntity(insportId, null, null);
+        NetworkElementEntity inspectedPort = new NetworkElementEntity();
+        inspectedPort.setElementId(entity.getInspectedPort().getElementId());
+
+        InspectionHookEntity retValEntity = this.txControl.required(() -> {
+            InspectionPortEntity dbInspectionPort = (InspectionPortEntity) this.api.getInspectionPort(inspectionPort);
+            this.utils.throwExceptionIfNullEntity(dbInspectionPort, inspectionPort);
+
+            InspectionHookEntity inspectionHookEntity = this.utils.findInspHookByInspectedAndPort(inspectedPort,
+                    dbInspectionPort);
+            if (inspectionHookEntity != null) {
+                String msg = String.format("inspection hook element exists for inspection id: %s, inspected id: %s\n",
+                        insportId, inspectedPort.getElementId());
+                LOG.error(msg);
+                throw new IllegalArgumentException(msg);
+            }
+
+            inspectionHookEntity = this.utils.makeInspectionHookEntity(inspectedPort, dbInspectionPort, entity.getTag(),
+                    entity.getEncType(), entity.getOrder(), entity.getFailurePolicyType());
+            inspectionHookEntity = this.em.merge(inspectionHookEntity);
+            return inspectionHookEntity;
+        });
+        return retValEntity.getHookId();
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport/{insportId}/inshook/{inshookId}")
+    @PUT
+    public InspectionHookEntity updateInspectionHook(@PathParam("ipAddr") String ipAddr,
+            @PathParam("insportId") String insportId, @PathParam("insportId") String inshookId,
+            InspectionHookEntity entity) throws Exception {
```

Comment:
Same here, can you simply call the updateInspectionhook from the SDK implementation?

Reply:
ok